### DISCUSSION TITLE
(Duckstation) Reverse Version Detection Regex on HTML Source

### DISCUSSION THREAD

**Discussion by SoumyBhow** [ImranR98/Obtainium#1231] [Q&A] [answered]
Hello. I'm trying to add Duckstation beta in Obtainium. I've added https://www.duckstation.org/android/ as the source, https://www.duckstation.org/android/duckstation-beta.apk as the custom APK link filter and \d\.\d-\d+ as the version extraction regex. While it's able to download the correct APK, the version it detects is the oldest one instead of the latest, and I can't find any options regarding reversing this. How can I have it detect the latest version? Thanks in advance.

**Comment by hyperpuncher** [comment]
You can now simply add https://www.duckstation.org/android/duckstation-beta.apk to the **App Source** and select **Direct APK Link** inside the **Override Source**.

**Reply by SoumyBhow** [reply]
That works. I was just hoping I could use the changelog from the original page to determine the version..

**Reply by hyperpuncher** [reply, marked as answer]
My bad, you can extract latest version by adding `(\d.\d-\d+).*` to **Version String Extraction RegEx**, `$1` to **Match Group** and enabling **Apply version string extraction Regex to entire page**.

**Reply by SoumyBhow** [reply]
That worked wonderfully. Thank you very much
